### PR TITLE
update p2p reconnect & test

### PR DESCRIPTION
**Pull request by MacOMNI**

### CODE REVIEW

File: Networking/Sources/Networking/Connection.swift [hunk 1]
```diff
@@ -100,6 +116,19 @@ public final class Connection<Handler: StreamHandler>: Sendable, ConnectionInfoP
...
                 true
+            case .reconnect:
+                false
+            }
+        }
+    }
+
+    public var needReconnect: Bool {
+        state.read {
+            switch $0 {
```
Comment (xlc) [inline Networking/Sources/Networking/Connection.swift:127]: `$0 == .reconnect`

File: Networking/Sources/Networking/Peer.swift [hunk 2]
```diff
@@ -259,6 +272,35 @@ final class PeerImpl<Handler: StreamHandler>: Sendable {
         }
     }
 
+    // TODO: Add reconnection attempts & Apply exponential backoff delay
+    func reconnect(to address: NetAddr, role: PeerRole) throws {
+        Task {
+            try await Task.sleep(for: .microseconds(2000))
```
Comment (xlc) [inline Networking/Sources/Networking/Peer.swift:278]: 2000 microseconds? so 2 miliseconds?

File: Networking/Sources/Networking/Peer.swift [hunk 3]
```diff
@@ -259,6 +272,35 @@ final class PeerImpl<Handler: StreamHandler>: Sendable {
...
+                )
+                try quicConn.connect(to: address)
+                let conn = Connection(
+                    quicConn,
+                    impl: self,
+                    role: role,
+                    remoteAddress: address,
+                    initiatedByLocal: true
+                )
+                connections.byAddr[address] = conn
```
Comment (xlc) [inline Networking/Sources/Networking/Peer.swift:298]: do we need to cleanup previous connection?

Comment (MacOMNI) [reply Networking/Sources/Networking/Peer.swift:298]: The connection resource has been cleaned up

File: Networking/Sources/Networking/Peer.swift (hunk 2 above)
Comment (MacOMNI) [reply Networking/Sources/Networking/Peer.swift:278]: Task will be removed

File: Networking/Sources/Networking/Connection.swift (hunk 1 above)
Comment (MacOMNI) [reply Networking/Sources/Networking/Connection.swift:127]: changed

**Review by xlc — APPROVED** [review]

**Pull request merged**